pkg/replication/sync_inspector.go

```diff
@@ -0,0 +1,225 @@
+							log.Warn().Msg("Destination syncmaster is configured for different source")
+							cancelSyncNeeded = true
+							if dr.status.Conditions.Update(api.ConditionTypeConfigured, false, "Invalid", "Destination syncmaster is configured for different source") {
+								updateStatusNeeded = true
+							}
+						}
+					}
+				} else {
+					// Destination has correct source, but is inactive
+					configureSyncNeeded = true
+					if dr.status.Conditions.Update(api.ConditionTypeConfigured, false, "Inactive", "Destination syncmaster is configured correctly but in-active") {
+						updateStatusNeeded = true
+					}
+				}
+			}
+
+			// Inspect source
+			sourceClient, err := dr.createSyncMasterClient(spec.Source)
+			if err != nil {
+				log.Warn().Err(err).Msg("Failed to create source syncmaster client")
+			} else {
+				sourceStatus, err := sourceClient.Master().Status(ctx)
+				if err != nil {
+					log.Warn().Err(err).Msg("Failed to fetch status from source syncmaster")
+				}
+
+				if sourceStatus.Status.IsActive() {
+					hasOutgoingEndpoint, err := dr.hasOutgoingEndpoint(sourceStatus, spec.Destination)
+					if err != nil {
+						log.Warn().Err(err).Msg("Failed to check has-outgoing-endpoint")
+					} else {
+						if hasOutgoingEndpoint {
+							// Source is correctly configured
+						}
+					}
+				}
+			}
+
+			// Update status if needed
+			if updateStatusNeeded {
+				if err := dr.updateCRStatus(); err != nil {
+					log.Warn().Err(err).Msg("Failed to update status")
+					hasError = true
+				}
+			}
+
+			// Cancel sync if needed
+			if cancelSyncNeeded {
+				req := client.CancelSynchronizationRequest{}
+				log.Info().Msg("Canceling synchronization")
+				if _, err := destClient.Master().CancelSynchronization(ctx, req); err != nil {
+					log.Warn().Err(err).Msg("Failed to cancel synchronization")
+					hasError = true
+				} else {
+					log.Info().Msg("Canceled synchronization")
+					nextInterval = time.Second * 10
+				}
+			}
+
+			// Configure sync if needed
+			if configureSyncNeeded {
+				source, err := dr.createArangoSyncEndpoint(spec.Source)
+				if err != nil {
+					log.Warn().Err(err).Msg("Failed to create syncmaster endpoint")
+					hasError = true
+				} else {
+					auth, err := dr.createArangoSyncTLSAuthentication(spec)
+					if err != nil {
+						log.Warn().Err(err).Msg("Failed to configure synchronization authentication")
+						hasError = true
+					} else {
+						req := client.SynchronizationRequest{
+							Source:         source,
+							Authentication: auth,
+						}
+						log.Info().Msg("Configuring synchronization")
+						if err := destClient.Master().Synchronize(ctx, req); err != nil {
+							log.Warn().Err(err).Msg("Failed to configure synchronization")
+							hasError = true
+						} else {
+							log.Info().Msg("Configured synchronization")
+							nextInterval = time.Second * 10
+						}
+					}
+				}
+			}
+		}
+	}
+
+	// Update next interval (on errors)
+	if hasError {
```

Comment:
there are several "err != nil" that generate log messages but do not set hasError.  Should it be set in any of those cases?